x/gov/client/cli/tx.go

```diff
@@ -0,0 +1,237 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	authcmd "github.com/cosmos/cosmos-sdk/x/auth/client/cli"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/pkg/errors"
+)
+
+const (
+	flagProposalID   = "proposalID"
+	flagTitle        = "title"
+	flagDescription  = "description"
+	flagProposalType = "type"
+	flagDeposit      = "deposit"
+	flagProposer     = "proposer"
+	flagDepositer    = "depositer"
+	flagVoter        = "voter"
+	flagOption       = "option"
+)
+
+// submit a proposal tx
+func GetCmdSubmitProposal(cdc *wire.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "submitproposal",
+		Short: "Submit a proposal along with an initial deposit",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			title := viper.GetString(flagTitle)
+			description := viper.GetString(flagDescription)
+			proposalType := viper.GetString(flagProposalType)
+			initialDeposit := viper.GetString(flagDeposit)
+
+			// get the from address from the name flag
+			from, err := sdk.GetAccAddressBech32(viper.GetString(flagProposer))
+			if err != nil {
+				return err
+			}
+
+			amount, err := sdk.ParseCoins(initialDeposit)
+			if err != nil {
+				return err
+			}
+
+			// create the message
+			msg := gov.NewMsgSubmitProposal(title, description, proposalType, from, amount)
+			// build and sign the transaction, then broadcast to Tendermint
+			ctx := context.NewCoreContextFromViper().WithDecoder(authcmd.GetAccountDecoder(cdc))
+
+			res, err := ctx.EnsureSignBuildBroadcast(ctx.FromAddressName, msg, cdc)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("Committed at block:%d. Hash:%s.Response:%+v \n", res.Height, res.Hash.String(), res.DeliverTx)
+			return nil
+		},
+	}
+
+	cmd.Flags().String(flagTitle, "", "title of proposal")
+	cmd.Flags().String(flagDescription, "", "description of proposal")
+	cmd.Flags().String(flagProposalType, "", "proposalType of proposal")
+	cmd.Flags().String(flagDeposit, "", "deposit of proposal")
+	cmd.Flags().String(flagProposer, "", "proposer of proposal")
+
+	return cmd
+}
+
+// set a new Deposit transaction
+func GetCmdDeposit(cdc *wire.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deposit",
+		Short: "deposit your token [steak] for activing proposal",
```

Comment:
shouldn't hard code in `steak` instead just use a generic term like `bonding-token`

Reply:
It can be something different than the bonding token, so I'll just leave it open